centreon/src/App/ResourceConfiguration/Infrastructure/ApiPlatform/State/ListPluginsProvider.php

```
@@ -0,0 +1,73 @@
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace App\ResourceConfiguration\Infrastructure\ApiPlatform\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProviderInterface;
+use App\ResourceConfiguration\Domain\Aggregate\Option;
+use App\ResourceConfiguration\Domain\Aggregate\OptionName;
+use App\ResourceConfiguration\Domain\Aggregate\Plugin;
+use App\ResourceConfiguration\Domain\Exception\PluginPathOptionDoesNotExistsException;
+use App\ResourceConfiguration\Domain\Repository\OptionRepository;
+use App\ResourceConfiguration\Domain\Repository\PluginRepository;
+use App\ResourceConfiguration\Infrastructure\ApiPlatform\Resource\PluginResource;
+use App\ResourceConfiguration\Infrastructure\ApiPlatform\Transformer\PluginTransformer;
+use App\Shared\Infrastructure\ApiPlatform\Transformer\TransformerInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+
+/**
+ * @implements ProviderInterface<PluginResource>
+ */
+final readonly class ListPluginsProvider implements ProviderInterface
+{
+    /**
+     * @param TransformerInterface<Plugin,PluginResource> $transformer
+     */
+    public function __construct(
+        #[Autowire(service: PluginTransformer::class)]
+        private TransformerInterface $transformer,
+        private PluginRepository $pluginRepository,
+        private OptionRepository $optionRepository,
+    ) {
+
+    }
+
+    /**
+     * @return iterable<PluginResource>
+     */
+    public function provide(Operation $operation, array $uriVariables = [], array $context = []): iterable
+    {
+        $pluginPathOption = $this->optionRepository->findByName(new OptionName(Option::PLUGIN_PATH_OPTION_NAME));
+        if (! $pluginPathOption instanceof Option) {
+            throw new PluginPathOptionDoesNotExistsException(['name' => Option::PLUGIN_PATH_OPTION_NAME]);
+        }
```

DIdn't we agree on using getXXX in repository when we want to throw an exception?

---

We should discuss about it, I was asking myself about it, because the exception is IMO related to this use case, not the existence of the data.